tests/unittests/Foundation/NSOrderedSetTests.mm

```diff
@@ -0,0 +1,324 @@
+//******************************************************************************
+//
+// Copyright (c) 2015 Microsoft Corporation. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <Foundation/Foundation.h>
+#import <TestFramework.h>
+
+void assertOrderedSetContent(NSOrderedSet* set, NSObject* first, ...) {
```

Comment:
The test wouldn't depend on another test. Both would depend on a common helper function.

In terms of code hygiene and maintainability, we want to factor out duplicated code where possible. Imagine not that someone wants to add different logic in the future (as this is always easy by adding a new function and calling it) but instead imagine someone changing the logic to an existing function. With two functions of duplicated logic, this makes two places the author needs to change and increases the opportunity for bugs / mismatches to arise. 

Renaming the functions is not sufficient as the extern / one definition rule issue still exists if someone else comes along and creates their own function called this. We should be very precise about how we expose functions where possible. Either a function is private to that TU and should be marked as static or it is (possibly) shared across files and it is not. Here you are claiming they are private but only changed the name. 

Please change this to have a common helper .h file (along with an implementation .mm or .cpp) that each test includes for this function.